### PR TITLE
[Tests] Use file `Data` instead of `String` for `runParseTest`

### PR DESCRIPTION
**Pull request by rintaro**
We should test invalid UTF-8 sequence handling

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please test

File: Tests/SwiftParserTest/ParserTests.swift [hunk 1]
```diff
@@ -6,9 +6,13 @@ import SwiftParser
...
   /// Run a single parse test.
   func runParseTest(fileURL: URL, checkDiagnostics: Bool) throws {
-    let fileContents = try String(contentsOf: fileURL)
-    let parsed = try Parser.parse(source: fileContents)
-    AssertStringsEqualWithDiff("\(parsed)", fileContents)
+    let fileContents = try Data(contentsOf: fileURL)
+    let parsed = try fileContents.withUnsafeBytes({ buffer in
+      try Parser.parse(source: buffer.bindMemory(to: UInt8.self))
+    })
+    // FIXME: This should compare binaries for handling invalid UTF-8 sequences.
```
Comment (DougGregor) [inline Tests/SwiftParserTest/ParserTests.swift:13]: Use the new "syntaxTextBytes" here!

Comment (rintaro) [reply Tests/SwiftParserTest/ParserTests.swift]: Introduced `AssertDataEqualWithDiff`. How does this look?

**Review by DougGregor — APPROVED** [review]

File: Tests/SwiftParserTest/ParserTests.swift (hunk 1 above)
Comment (DougGregor) [reply Tests/SwiftParserTest/ParserTests.swift]: LGTM.  There are a handful of disabled tests in https://github.com/apple/swift-syntax/pull/764 that could be enabled by your change here.

Comment (rintaro) [reply Tests/SwiftParserTest/ParserTests.swift]: I will rebase this PR and try enabling them after #764 lands

**Comment by rintaro** [comment]
@swift-ci Please test

File: Tests/SwiftParserTest/ParserTests.swift [hunk 2]
```diff
@@ -95,11 +97,6 @@ public class ParserTests: XCTestCase {
         // This test causes an assertion in the string literal lexer
         || fileURL.absoluteString.contains("string_literal_eof3.swift")
 
-        // These tests have invalid UTF-8 in the source files and are not
-        // properly checked.
-        || fileURL.absoluteString.contains("invalid-utf8.swift")
```
Comment (DougGregor) [inline Tests/SwiftParserTest/ParserTests.swift:100]: Wonderful!

**Pull request merged**